### ISSUE TITLE
[bitnami/openldap] Need some help in enabling password policy module in openldap

### ISSUE THREAD

**Issue by rohan-97** [bitnami/containers#6671]
### Name and Version

bitnami/openldap:2.6.3-debian-11-r16

### What steps will reproduce the bug?

Hello,

First of all thanks for the highly secure and robust OpenLDAP container!  
I am trying to create OpenLDAP docker container by adding custom schema files.  

One of the requirements is to have password policy enabled in the OpenLDAP.

I checked for password policy in existing schemas inside the container and was not able to find one.
Hence I was looking for password policy schema and I found that the OpenLDAP Debian package ships password policy schema by default.

Hence I downloaded debian package from [here}(https://packages.debian.org/bullseye/slapd) and extracted the debian package.

I found following files
```
 - /etc/ldap/schema/ppolicy.ldif
 - /etc/ldap/schema/ppolicy.schema
```
above 2 files were added to the following path inside container  
`/opt/bitnami/openldap/etc/schema/`

I used following command to create docker image

```
docker run -d --env LDAP_ADMIN_USERNAME=admin   --env LDAP_ADMIN_PASSWORD=adminpassword   --env "LDAP_ROOT=dc=testdomain,dc=com" --env "LDAP_EXTRA_SCHEMAS=cosine,ppolicy,nis,inetorgperson" --env "BITNAMI_DEBUG=true" -v /root/test_ldif_files/:/ldiff test-openldap
```

I ran following command to confirm whether ppolicy schema was loaded or not
```
ldapsearch -LLLQY EXTERNAL -H ldapi:/// -b cn=schema,cn=config "(objectClass=olcSchemaConfig)" dn
```

and following is the output of above command
```
I have no name!@32ac857f2fa9:/$ ldapsearch -LLLQY EXTERNAL -H ldapi:/// -b cn=schema,cn=config "(objectClass=olcSchemaConfig)" dn
dn: cn=schema,cn=config

dn: cn={0}core,cn=schema,cn=config

dn: cn={1}cosine,cn=schema,cn=config

dn: cn={2}ppolicy,cn=schema,cn=config

dn: cn={3}nis,cn=schema,cn=config

dn: cn={4}inetorgperson,cn=schema,cn=config
```

after confirming that ppolicy file was loaded successfully,
I tried to perform steps mentioned over here

http://tutoriels.meddeb.net/openldap-password-policy-managing-users-accounts/

when I tried to add following LDIF file
```
dn: cn=module{0},cn=config
changeType: modify
add: olcModuleLoad
olcModuleLoad: ppolicy
```
using the following command
```
ldapmodify  -H 'ldapi:///' -D "cn=admin,dc=testdomain,dc=com" -w "adminpassword" -f /ldiff/test.ldif
```

I get the following error
```
I have no name!@32ac857f2fa9:/$ ldapmodify  -H 'ldapi:///' -D "cn=admin,dc=testdomain,dc=com" -w "adminpassword" -f /ldiff/test.ldif
modifying entry "cn=module{0},cn=config"
ldap_modify: No such object (32)
        matched DN: cn=config
```

I have attached the following schema files which I got from the OpenLDAP Debian package
[ppolicy.schema.txt](https://github.com/bitnami/containers/files/9582564/ppolicy.schema.txt)
[ppolicy.ldif.txt](https://github.com/bitnami/containers/files/9582566/ppolicy.ldif.txt)

can you let me know if the above schema files are the correct ones to load ppolicy?  
if not then can you specify the recommended ppolicy schema files?  

also, kindly let me know if any of the step is missing/wrong.

Thanks

### What is the expected behavior?

password policy should be added and applied successfully.

### What do you see instead?

error in enabling password policy

**Comment by fmulero** [comment]
Hi @rohan-97 

In this [comment](https://github.com/bitnami/containers/issues/982#issuecomment-1220354408) you have information about how to enable a module in our container. Please pay attention to the paths and module name (with the extension `.so`) in the `ldif` file:
```ldif
dn: cn=module,cn=config
cn: module
objectClass: olcModuleList
olcModulePath: /opt/bitnami/openldap/lib/openldap
olcModuleLoad: memberof.so
olcModuleLoad: refint.so
```

**Comment by rohan-97** [comment]
Hi @fmulero ,  
Thanks for the answer,

I searched for ppolicy.so file in the container and found the file present with the following path.
```
/opt/bitnami/openldap/lib/openldap/ppolicy.so
```

so I created following ldif file as per above example.
```
dn: cn=module,cn=config
cn: module
objectClass: olcModuleList
olcModulePath: /opt/bitnami/openldap/lib/openldap
olcModuleLoad: ppolicy.so
```

and tried to apply the ldif file using following command.
```
ldapadd -H 'ldapi:///' -D "cn=admin,dc=testdomain,dc=com" -w "adminpassword" -f /ldiff/test.ldif
```

and I am getting insufficient access as per following output
```
I have no name!@62de44e5d59d:/$ ldapadd -H 'ldapi:///' -D "cn=admin,dc=testdomain,dc=com" -w "adminpassword" -f /ldiff/test.ldif
adding new entry "cn=module,cn=config"
ldap_add: Insufficient access (50)
```
Note: I am using the user which was created using `LDAP_ADMIN_USERNAME` environment variable.  

I also tried to add this ldif under `LDAP_CUSTOM_LDIF_DIR` directory but container failed to start with same error as mentioned above.

can you please let me know whether am I missing something here or do I need to setup separate user for loading password policy?

Thanks in advance

**Comment by fmulero** [comment]
If you mount the file in `/opt/bitnami/openldap/etc/schema/ppolicy.ldif` and add this new schema to the variable `LDAP_EXTRA_SCHEMAS:.....,ppolicy` as you can see in [the previous example](https://github.com/bitnami/containers/issues/982#issuecomment-1220354408) you don't need to add it manually, the container will do it during start up.

**Comment by rohan-97** [comment]
Hey @fmulero ,  

I followed the steps provided by you and removed the ppolicy.ldif file which I got from openldap debian package and replaced it with following content

```
dn: cn=module,cn=config
cn: module
objectClass: olcModuleList
olcModulePath: /opt/bitnami/openldap/lib/openldap
olcModuleLoad: ppolicy.so
```

and it worked!

I was able to add passwordpolicy to the users...
Thanks a lot for guiding through the steps!
the example that you shared was very helpful.

Thanks a lot for the quick help 😊
